utils/compression.sql

```diff
@@ -0,0 +1,42 @@
+-- Update metadata for chunks compressed before 2.0.
+--
+-- After you have updated from pre-2.0 it might be that some
+-- compressed chunks are missing information about the approximate row
+-- count. To correct that, call this function. Note that it will only
+-- update approximate row counts that are NULL, so running it several
+-- times will not overwrite existing values.
+--
+-- To avoid blocking too many tables, it will commit after each chunk,
+-- which means that the function might be slow if there is a lot of
+-- data to update, but it will not block other queries.
+CREATE PROCEDURE correct_compression_chunk_size ()
+LANGUAGE PLPGSQL
+AS $$
+DECLARE
+  plain_chunk RECORD;
+  comp_chunk TEXT;
+  rowcount_pre BIGINT;
+  rowcount_post BIGINT;
+BEGIN
+  FOR plain_chunk IN
+      SELECT * FROM _timescaledb_catalog.chunk WHERE compressed_chunk_id IS NOT NULL
+  LOOP
+      SELECT format('%I.%I', schema_name, table_name) INTO comp_chunk
+      FROM _timescaledb_catalog.chunk
+      WHERE id = plain_chunk.compressed_chunk_id;
```

Comment:
I think this should start from the `compression_chunk_size` catalog and join from there, and there should be a where clause on `numrows_post_compression IS NULL OR numrows_pre_compression IS NULL`  so that we only get those chunks that have a null in one of those to try and populate, this could happen if you run this later after new chunks have been created in 2.0 or if you ran it and canceled it but because it's not transactional, it only got part way through. This then means you're not doing all the extra work of scanning the chunks where you're not going to update later.

Reply:
sorry I guess this should actually be the select statement driving the loop.

Reply:
Yeah, I thought it looked a little backwards as well but since this had gone through the hoops (been reviewed already) I was unsure if I missed something. Good to see that I didn't. I have basically inverted it and picked out the rows that need to be fixed and then figure out what relation to select from.